x-pack/plugins/fleet/public/applications/fleet/sections/epm/screens/detail/index.test.tsx

```diff
@@ -190,12 +224,27 @@ describe('when on integration detail', () => {
   });
 });
 
-interface MockedApi {
+interface MockedApi<
```

Comment:
Some of this here is refactoring which will facilitate breaking this out to a separate set of test utilities so that this approach can be used with other test areas of Fleet.
The mocked api interface could now also return the set of functions that are used to fulfill the API calls, which allow each test case to adjust them (if necessary) before rendering the UI.